src/Avalonia.Themes.Fluent/Controls/SplitButton.xaml

```diff
@@ -0,0 +1,205 @@
+    <Setter Property="BorderThickness" Value="{DynamicResource SplitButtonBorderThemeThickness}" />
+    <Setter Property="HorizontalAlignment" Value="Left" />
+    <Setter Property="VerticalAlignment" Value="Center" />
+    <!--<Setter Property="UseSystemFocusVisuals" Value="True" />
+    <Setter Property="FocusVisualMargin" Value="-3" />-->
+    <Setter Property="KeyboardNavigation.IsTabStop" Value="True" />
+    <Setter Property="Focusable" Value="True" />
+    <Setter Property="Padding" Value="{DynamicResource ButtonPadding}" />
+    <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate>
+          <Grid>
+            <Grid.ColumnDefinitions>
+              <ColumnDefinition Width="*" />
+              <ColumnDefinition Width="Auto" />
+            </Grid.ColumnDefinitions>
+
+            <Button x:Name="PART_PrimaryButton"
+                    Grid.Column="0"
+                    MinWidth="{DynamicResource SplitButtonPrimaryButtonSize}"
+                    Foreground="{TemplateBinding Foreground}"
+                    Background="{TemplateBinding Background}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Content="{TemplateBinding Content}"
+                    ContentTemplate="{TemplateBinding ContentTemplate}"
+                    Command="{TemplateBinding Command}"
+                    CommandParameter="{TemplateBinding CommandParameter}"
+                    FontFamily="{TemplateBinding FontFamily}"
+                    FontSize="{TemplateBinding FontSize}"
+                    FontWeight="{TemplateBinding FontWeight}"
+                    HorizontalAlignment="Stretch"
+                    VerticalAlignment="Stretch"
+                    HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
+                    VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
+                    Padding="{TemplateBinding Padding}"
+                    Focusable="False"
+                    KeyboardNavigation.IsTabStop="False" />
+
+            <Button x:Name="PART_SecondaryButton"
+                    Grid.Column="1"
+                    MinWidth="{DynamicResource SplitButtonSecondaryButtonSize}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Foreground="{TemplateBinding Foreground}"
+                    Background="{TemplateBinding Background}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    HorizontalContentAlignment="Right"
+                    VerticalContentAlignment="Center"
+                    HorizontalAlignment="Stretch"
+                    VerticalAlignment="Stretch"
+                    Focusable="False"
+                    KeyboardNavigation.IsTabStop="False" />
```

Comment:
Comment made here: https://github.com/AvaloniaUI/Avalonia/pull/7422#issuecomment-1021911616

 > You don't need PrimaryBackgroundRectangle, because you can directly modify Button's background depending on selector. So it can be removed. Same for SecondaryBackgroundRectangle.

Alright, makes sense. I'm not sure why upstream did it the way they did but that was copied over without much thought. This was more of a port before you made it a complete re-write.

Reply:
@maxkatz6 I discovered partly why this was done in upstream Fluent.

https://github.com/microsoft/microsoft-ui-xaml/blob/8d594493ca143c5724b8c84553eff1a16744321f/dev/SplitButton/SplitButton_rs1.xaml#L24

The PrimaryButton and SecondaryButton themselves are used only for hit targets basically. The actual style changes are applied to the PrimaryBackgroundRectangle, SecondaryBackgroundRectangle, Border (top) and RootGrid (bottom). Note that one Border is ON TOP in z-index and spans all 3 columns, the other RootGrid is on the bottom of the z-index and still spans all three columns. This means the Border/RootGrid itself goes around the entire control which is actually used in some states.

Now, it is still possible to remove the PrimaryBackgroundRectangle and SecondaryBackgroundRectangle but the Buttons CANNOT handle the border. This still needs to be done for the entire control at once or it won't properly manage the separator gap. This means you have to add back a Border control. However, you can place it on top of the buttons by adding it at the end of the XAML (as you know). That should still allow you to use the buttons for changing primary/secondary backgrounds without having the extra rectangles.

Note: Keep in mind that the entire control background also should change in some situations (instead of just Primary/Secondary button backgrounds). Obviously you can't use the same border for both border and background or it would entirely occlude the buttons due to z-index. This is why there is the RootBorder as well as the Border for actual borders on top. This will be discussed separately above.

This type of thing is why such drastic changes shouldn't be done without discussion first. I will leave this fix to you if you don't mind.

Reply:
Actually, putting the border on top of the buttons will make the buttons non-clickable for the thickness of the border. So that's probably why they made the buttons fully transparent on top just as hit-targets. Seems minor but that 1px border not being clickable might be considered a bug in the future.

Reply:
Yes, HitTestVisible would fix that, but I don't agree with some other reasons, why we need this border at all.
Will continue in another thread above.